Trio/Sources/Modules/Export/ExportStateModel.swift

```diff
@@ -0,0 +1,1514 @@
+        // Export formats for different file types
+        enum ExportFormat: String, CaseIterable, Identifiable {
+            var id: String { rawValue }
+            case csv
+            case json
+
+            var fileExtension: String {
+                switch self {
+                case .csv:
+                    return "csv"
+                case .json:
+                    return "json"
+                }
+            }
+        }
+
+        // Published state for UI binding
+        @Published var selectedCategories: Set<ExportCategory> = Set(ExportCategory.allCases)
+        @Published var selectedFormat: ExportFormat = .csv
+        @Published var isExporting: Bool = false
+
+        enum ExportError: LocalizedError {
+            case documentsDirectoryNotFound
+            case fileWriteError(Error)
+            case unknown(String)
+
+            var errorDescription: String? {
+                switch self {
+                case .documentsDirectoryNotFound:
+                    return String(localized: "Could not access documents directory")
+                case let .fileWriteError(error):
+                    return String(localized: "Failed to write export file: \(error.localizedDescription)")
+                case let .unknown(message):
+                    return String(localized: "Export failed: \(message)")
+                }
+            }
+        }
+
+        /// Exports selected Trio settings to a CSV file
+        ///
+        /// This function creates an export of the user's selected Trio configuration categories including:
+        /// - Export metadata (date, app version, build) [optional]
+        /// - Device settings (CGM, pump information) [optional]
+        /// - Therapy profiles (basal rates, ISF, carb ratios, targets) [optional]
+        /// - Algorithm settings (SMB, autosens, dynamic settings, etc.) [optional]
+        /// - Features and UI preferences [optional]
+        /// - Notification settings [optional]
+        /// - Service configurations [optional]
+        /// - Preset data [optional]
+        ///
+        /// - Parameter categories: Set of categories to include in export. If nil, exports all categories.
+        /// - Parameter format: Export format to use. If nil, uses currently selected format.
+        /// - Returns: A Result containing either the file URL on success or an ExportError on failure
+        func exportSettings(
+            categories: Set<ExportCategory>? = nil,
+            format: ExportFormat? = nil
+        ) async -> Result<URL, ExportError> {
+            debug(.default, "🔄 EXPORT: Starting settings export...")
+
+            let categoriesToExport = categories ?? selectedCategories
+            let exportFormat = format ?? selectedFormat
+            debug(
+                .default,
+                "🔄 EXPORT: Exporting categories: \(categoriesToExport.map(\.rawValue).joined(separator: ", ")) in \(exportFormat.rawValue) format"
+            )
+
+            debug(
```

Comment:
CoreData should be ready to go -- it won't load the app unless it is. Have you seen failures in testing with it?